hal/src/nRF52840/gsm0710muxer/channel_stream.h

```diff
@@ -217,10 +217,11 @@ inline int MuxerChannelStream<MuxerT>::waitEvent(unsigned flags, unsigned timeou
         if (f &= flags) {
             break;
         }
-        if (timeout > 0 && HAL_Timer_Get_Milli_Seconds() - t >= timeout) {
+        const auto elapsed = HAL_Timer_Get_Milli_Seconds() - t;
+        if (timeout == 0 || elapsed >= timeout) {
             return SYSTEM_ERROR_TIMEOUT;
         }
-        os_semaphore_take(sem_, HAL_Timer_Get_Milli_Seconds() - t, false);
+        os_semaphore_take(sem_, timeout - elapsed, false);
```

Comment:
What if `timeout == 0` ?  Will the `os_semaphore_take()` call handle this?

Reply:
If it's zero, we won't reach this call at all an will return `SYSTEM_ERROR_TIMEOUT` above.